UC_NCGO_Fall/Assets/App/Resource/Scripts/UI/NetworkedPlayerData.cs

```diff
@@ -0,0 +1,125 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Unity.Netcode;
+
+public class NetworkedPlayerData : NetworkBehaviour
+{
+   public NetworkList<PlayerInfoData> _allConnectedPlayers; // current players in the game
+   private int _players = -1;
+   private ulong _serverLocalID;
+
+   private Color[] _PlayerColors = new Color[]
+   {
+    Color.blue, Color.red, Color.magenta, Color.yellow, Color.white, Color.black, Color.green
+   };
+
+   private void Awake()
+   {
+    //avoid mem leaks by ini network list here
+    _allConnectedPlayers = new NetworkList<PlayerInfoData>(readPerm: NetworkVariableReadPermission.Everyone);
+    //shouldn't need to dispose of this but keep getting an error?
+   }
+
+    public override void OnNetworkSpawn()
+    {
+        base.OnNetworkSpawn();
+
+        if(IsServer)
+        {
+            NetworkManager.Singleton.OnConnectionEvent += OnConnectionEvents;
+            _serverLocalID = NetworkManager.LocalClientId;
+    
+        }
+    }
+
+    public override void OnNetworkDespawn()
+    {
+        if(IsServer)
+        {
+            NetworkManager.Singleton.OnConnectionEvent -= OnConnectionEvents;
+        }
+        //clears the memory leak
+        if (_allConnectedPlayers != null)
+        {
+        _allConnectedPlayers.Clear();
+        }
+        base.OnNetworkDespawn();
+    }
+
+    public void OnConnectionEvents(NetworkManager netManager, ConnectionEventData eventData)
+    {
+        if(eventData.EventType == ConnectionEvent.ClientConnected)
+        {
+            //when client connects create data
+            CreateNewClientData(eventData.ClientId);
+        }
+        if(eventData.EventType == ConnectionEvent.ClientDisconnected)
+        {
+            RemovePlayerData(FindPlayerInfoData(eventData.ClientId));
+            _players--;
+        }
+    }
+    
+    
+
+   private void CreateNewClientData(ulong clientID){
+    // creating player info
+    PlayerInfoData _playerInfoData = new PlayerInfoData(clientID);
+
+    //add or modify name
+    //check to see if server matches parameter client id
+
+    
+    if(_serverLocalID == clientID)
+    {
+        _playerInfoData._isPlayerReady = false;
+    }
```

Comment:
` _playerInfoData._isPlayerReady = false; ` should be initialized as true. In this situation we want to assume the Host is always ready in the lobby,  so we want to sync that dat with the client ID matches the server. This will fix your Ready up logic that will turn on and off Start game.

Reply:
Will be resolved in latest update.